HIP/HIP-668.md

```diff
@@ -0,0 +1,2349 @@
+  """
+  expirationTimestamp: Timestamp
+
+  "An opaque, globally unique identifier specific to GraphQL."
+  id: ID!
+
+  """
+  The admin key associated with this entity whose signing requirements must be met in order to modify the entity on
+  the network. This returns a dynamic map that varies per entity and may be arbitrarily complex depending upon this
+  key's signing requirements.
+  """
+  key: Object
+
+  "The memo associated with the entity."
+  memo: String!
+
+  "The accountable entity to receive the remaining balance from the deleted entity."
+  obtainer: Accountable
+
+  """
+  The pending reward the account will receive in the next reward payout. Note the value is updated at the end of each
+  staking period and there may be delay to reflect the changes in the past staking period. Defaults to tinybars.
+  """
+  pendingReward(unit: HbarUnit = TINYBAR): Long
+
+  "The entity to which this account is staked. Mutually exclusive with stakedNode."
+  stakedEntity: Accountable
+
+  "The node to which this account is staked. Mutually exclusive with stakedEntity."
+  stakedNode: NetworkNode
+
+  """
+  The staking period during which either the staking settings for this account changed (such as starting staking or
+  changing stakedNode) or the most recent reward was earned, whichever is later. If this account is not currently
+  staked to a node, then the value is null.
+  """
+  stakePeriodStart: Timestamp
+
+  "A consensus timestamp range with an inclusive from timestamp and an exclusive to timestamp."
+  timestamp: TimestampRange!
+
+  "The type of Hedera entity."
+  type: EntityType!
+}
+
+"The encoding used to convert the raw bytes to String."
+enum BinaryEncoding {
+  "Base 64 encoding."
+  BASE64
+
+  "Base 16 hexadecimal encoding with a '0x' prefix and using lowercase letters and numbers."
+  HEX
+}
+
+"""
+The base type for an entity on the Hedera network. Each entity is assigned an incrementing identifier that is shared
+across all entity types and associated with a particular shard and realm.
+"""
+interface Entity implements Node {
+  "The account charged the auto-renewal fee."
+  autoRenewAccount: Account
+
+  "The amount of time to elapse before auto-renew occurs."
+  autoRenewPeriod: Duration
+
+  "The consensus timestamp at which the entity was created."
+  createdTimestamp: Timestamp
+
+  "Whether the entity still exists on the ledger."
+  deleted: Boolean
+
+  "The unique identifier associated with the entity."
+  entityId: EntityId!
+
+  """
+  The time at which this entity will expire and be auto-renewed, if possible. If this was not explicitly set by the
+  user it will be calculated as the createdTimestamp plus the autoRenewPeriod.
+  """
+  expirationTimestamp: Timestamp
+
+  "An opaque, globally unique identifier specific to GraphQL."
+  id: ID!
+
+  """
+  The admin key associated with this entity whose signing requirements must be met in order to modify the entity on
+  the network.
+  """
+  key: Object
```

Comment:
This dynamic structure is a bit unique. Need to document it separately somewhere.